action/devices.py

```diff
@@ -84,47 +95,151 @@ def __call__(self, args: list[str]) -> list[str]:
 
         assert len(args) >= 1, "not enough parameters passed"
 
-        return [args[0]]
+        if type(args[0]) is int:
+            return [format(args[0], "#x")]
+        else:
+            return [args[0]]
 
     def check_args(self, args: list[str]) -> bool:
-        return len(args) == 1 and \
-               len(args[0]) >= 3 and \
-               args[0][0:2] == '0x' and \
-               all(c in hexdigits for c in args[0][2:]) and \
-               3 <= int(args[0], 16) <= 119
+
+        if len(args) != 1:
+            return False
+
+        if type(args[0]) is int:
+            return 3 <= args[0] <= 119
+        elif type(args[0]) is str:
+            return len(args[0]) >= 3 and \
+                   args[0][0:2] == '0x' and \
+                   all(c in hexdigits for c in args[0][2:]) and \
+                   3 <= int(args[0], 16) <= 119
+        else:
+            return False
+
+
+class DeviceType(Enum):
```

Comment:
Sounds like the type of the actual device. How about `DeviceRepr` or `DeviceStyle` or something?

Also, do we really need this? We could just translate the multiline string version to a dict in `get_device`.